app/pyassembler/assembler/assembler.py

```diff
@@ -11,36 +15,54 @@ def __init__(self):
         self.__delimiters = ['(', ')', ',']
         self.__symboltable = {}
 
-    def parse(self, code, commands, registers):
+    def parse(self, code_address, code, zeros_constants, commands, registers):
         bitcode = []
 
-        code = self.__generateSymboltable(code)
+        zeros_constants = self.__generate_start(
+            code_address,
+            zeros_constants,
+            commands,
+            registers
+        )
+        code = self.__generate_symboltable(code)
 
         for line_nr, line in enumerate(code):
-            LOGGER.debug(str(line))
+            logging.debug(str(line))
             for delimiter in self.__delimiters:
                 line = line.replace(delimiter, ' ')
             line = line.strip()
             tokens = line.split(' ')
             if len(tokens[0]) == 0:
                 continue
-            elif tokens[0] in commands['arithmetic']:
-                bitcode = bitcode + self.__parseArithmetic(
+            if tokens[0] in commands['arithmetic']:
+                bitcode = bitcode + self.__parse_arithmetic(
                     tokens, commands['arithmetic'], registers)
             elif tokens[0] in commands['storage']:
-                bitcode = bitcode + self.__parseStorage(
+                bitcode = bitcode + self.__parse_storage(
                     tokens, commands['storage'], registers)
             elif tokens[0] in commands['branch']:
                 bitcode = bitcode + \
-                    self.__parseBranch(
+                    self.__parse_branch(
                         line_nr, tokens, commands['branch'], registers)
             else:
                 raise Exception(
                     "Parsing error. Command not found: " + tokens[0])
 
-        return bitcode
-
-    def __generateSymboltable(self, code):
+        machine_code = self.__generate_machinecode(
+            code_address,
+            bitcode,
+            zeros_constants
+        )
+        return machine_code
+
+    def __generate_machinecode(self, code_address, bitcode, zeros_constants):
+        index = int(code_address / 4)
+        machine_code = zeros_constants[:]
```

Comment:
Can u please replace this with the global variable

Reply:
fixed.